### PR TITLE
chore: change CI backend infra 

### PR DESCRIPTION
**Pull request by subhashish-devtron**
# Description
code changes for setting infra in backend for change CI feature

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [x] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/app/AppListingService.go [hunk 1]
```diff
@@ -877,7 +878,7 @@ func (impl AppListingServiceImpl) setIpAccessProvidedData(ctx context.Context, a
 		if ciPipeline != nil && ciPipeline.CiTemplate != nil && len(*ciPipeline.CiTemplate.DockerRegistryId) > 0 {
 			dockerRegistryId := ciPipeline.CiTemplate.DockerRegistryId
 			appDetailContainer.DockerRegistryId = *dockerRegistryId
-			if !ciPipeline.IsExternal || ciPipeline.ParentCiPipeline != 0 {
+			if (!ciPipeline.IsExternal || ciPipeline.ParentCiPipeline != 0) && ciPipeline.PipelineType != string(bean2.LINKED_CD) {
```
Comment (kripanshdevtron) [inline pkg/app/AppListingService.go:881]: create a method with struct `CiPipeline` for logic `ciPipeline.ParentCiPipeline != 0) && ciPipeline.PipelineType != string(bean2.LINKED_CD)`

File: pkg/pipeline/CiCdPipelineOrchestrator.go [hunk 2]
```diff
@@ -884,6 +895,15 @@ func (impl CiCdPipelineOrchestratorImpl) CreateCiConf(createRequest *bean.CiConf
 		if err != nil {
 			return nil, err
 		}
+		if createRequest.Artifact != nil {
```
Comment (kripanshdevtron) [inline pkg/pipeline/CiCdPipelineOrchestrator.go:898]: add comment here, in which case we would receive this

File: pkg/pipeline/BuildPipelineConfigService.go [hunk 3]
```diff
@@ -1267,6 +1279,321 @@ func (impl *CiPipelineConfigServiceImpl) UpdateCiTemplate(updateRequest *bean.Ci
...
+		return nil, err
+	}
+
+	if pipelineExists {
+		impl.logger.Errorw("pipeline name already exist", "err", err, "patch cipipeline name", request.CiPipeline.Name)
+		return nil, fmt.Errorf("pipeline name already exist")
+	}
+
+	if request.IsSwitchCiPipelineRequest() {
+		impl.logger.Debugw("handling switch ci pipeline")
```
[the remark this answers is not in the document]
Comment (gireesh-devtron) [reply pkg/pipeline/BuildPipelineConfigService.go:1501]: done

File: pkg/pipeline/BuildPipelineConfigService.go [hunk 4]
```diff
@@ -1267,6 +1279,321 @@ func (impl *CiPipelineConfigServiceImpl) UpdateCiTemplate(updateRequest *bean.Ci
...
+		return impl.handleSwitchCiPipeline(request, ciConfig)
+	}
+
+	impl.logger.Debugw("create patch request")
```
[the remark this answers is not in the document]
Comment (gireesh-devtron) [reply pkg/pipeline/BuildPipelineConfigService.go:1505]: deleted this log

File: pkg/pipeline/BuildPipelineConfigService.go [hunk 5]
```diff
@@ -1267,6 +1279,321 @@ func (impl *CiPipelineConfigServiceImpl) UpdateCiTemplate(updateRequest *bean.Ci
...
+		return nil, err
+	}
+	defer impl.ciPipelineRepository.RollbackTx(tx)
+	oldAppWorkflowMapping, err := impl.deleteOldPipelineAndWorkflowMappingBeforeSwitch(tx, ciPipeline, request.SwitchFromExternalCiPipelineId, request.UserId)
+	if err != nil {
+		impl.logger.Errorw("error in deleting old ci-pipeline and getting the appWorkflow mapping of that", "err", err, "userId", request.UserId)
+		return nil, err
+	}
+
+	impl.logger.Debugw("create patch request for ci switch")
```
[the remark this answers is not in the document]
Comment (gireesh-devtron) [reply pkg/pipeline/BuildPipelineConfigService.go:1555]: removed it

**Review by kripanshdevtron — APPROVED** [review]

**Pull request merged**